pkg/util/provider/machinecontroller/machineclass.go

```diff
@@ -201,16 +171,16 @@ func (c *controller) reconcileClusterMachineClass(ctx context.Context, class *v1
 func (c *controller) addMachineClassFinalizers(ctx context.Context, class *v1alpha1.MachineClass) error {
 	finalizers := sets.NewString(class.Finalizers...)
 	finalizers.Insert(MCMFinalizerName)
-	return c.updateMachineClassFinalizers(ctx, class, finalizers.List())
+	return c.updateMachineClassFinalizers(ctx, class, finalizers.List(), true)
 }
 
 func (c *controller) deleteMachineClassFinalizers(ctx context.Context, class *v1alpha1.MachineClass) error {
 	finalizers := sets.NewString(class.Finalizers...)
 	finalizers.Delete(MCMFinalizerName)
-	return c.updateMachineClassFinalizers(ctx, class, finalizers.List())
+	return c.updateMachineClassFinalizers(ctx, class, finalizers.List(), false)
 }
 
-func (c *controller) updateMachineClassFinalizers(ctx context.Context, class *v1alpha1.MachineClass, finalizers []string) error {
+func (c *controller) updateMachineClassFinalizers(ctx context.Context, class *v1alpha1.MachineClass, finalizers []string, addFinalizers bool) error {
```

Comment:
No need for `addFinalizers`. I think we should print the log on the caller side.

Reply:
we are directly returning at the caller side, printing there with an if {...} statement would look more untidy i think, thats why I did it here.